File: Shop/src/app/components/orders-add/Views/AddOrderView.ts
import { AddProductOrderView } from './AddProductOrderView';

export class AddOrderView {
    Id?: number;
    CustomerId: number;
    Status: string;
    date?: Date;
    Comment: string;
    ProductOrders: AddProductOrderView[];
  
    constructor (CustomerId: number, Status: string, Comment: string, ProductOrders: AddProductOrderView[]) {
      this.CustomerId = CustomerId;
      this.Status = Status;
      this.Comment = Comment;
      this.ProductOrders = ProductOrders;

      // empty data
      this.Id = 0;
      this.date = new Date();
    }
}
  

File: WebShop/WebShop/Views/Add/AddOrderView.cs
namespace WebShop.Views.Add
{
    public class AddOrderView
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public string Status { get; set; }
        public string Comment { get; set; }
        public DateTime Date { get; set; }
        public List<AddProductOrderView> ProductOrders { get; set; }
    }
}
